README
README for MySQL-Diff (guyfawkes)

                              MySQL-Diff
                             ============

MySQL-Diff is suite of Perl modules and accompanying CLI script for
comparing the schema (table structures) of two MySQL databases.

Prerequisites
-------------

This suite requires Perl 5.006 or higher.

Availability
------------

The latest version of MySQL-Diff is available from

  http://adamspiers.org/computing/mysqldiff/

and eventually (a possibly slightly older version) from the
Comprehensive Perl Archive Network (CPAN).  Visit
<http://www.perl.com/CPAN> to find a CPAN site near you.

Installation
------------

See INSTALL file.

Documentation
-------------

Homepage: http://adamspiers.org/computing/mysqldiff/
Documentation at CPAN: http://search.cpan.org/dist/MySQL-Diff/

Support
-------

Questions and suggestions for MySQL-Diff should just be sent
to me at <mysqldiff@adamspiers.org>.
Patches should be sent as pull requests to http://github.com/aspiers/mysqldiff
See https://github.com/aspiers/mysqldiff/wiki
Bug reports should be submitted at the website below.

Known bugs
----------

See https://rt.cpan.org/Public/Dist/Display.html?Name=MySQL-Diff

Copyright
---------

(c) 2000-2011 Adam Spiers <mysqldiff@adamspiers.org>, all rights reserved.
This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.


Changes
# Changes log for Test::CPAN::Meta

0.51    23th April 2014
        - Merge many bugfixes and extensions by guyfawkes (Vitaly Evsyukov):
          - multiple fixes with primary keys
          - parses correctly stored procedures, functions and triggers
          - added parameter --refs (-r). Now you can view all refs in database
          - add support for foreign keys and views
          - fix for TIMESTAMP columns statements order, also for AFTER clause
          - multiple bug fixes with columns order, fields diff, primary key, when not exists
            in one of two tables
          - added MySQL::Diff::Table public methods: fields_links, fields_order, fk_tables, foreign_key,
            get_fk_by_col, isa_fk
          - added MySQL::Diff::Utils public methods: get_logdir, save_logdir, write_log

0.44    22th April 2014
        - optimized the --table-re option to exclude
          tables before comparing them as this would
          cause significant slowdowns in databases
          with lots of tables. (bdraco)

0.43    6th October 2011
        - fix missing fields in CPAN meta-data

0.43    6th October 2011
        - depend on Perl 5.6
        - improve docs and CPAN meta-data

0.41    5th October 2011
        - tidy up POD

0.40    5th October 2011
        - fix issue with hyphens in database names
        - made --tolerant ignore COLLATE and AUTO_INCREMENT
        - fixed 'Duplicate specification' options from Getopt::Long
        - made CLI options case-sensitive
        - fixed some coding style inconsistencies
        - remove .cvsignore
        - merged changes by Barbie
          - removed use of unmaintained Class::MakeMethods
          - repackaged distribution with additional package files
          - restructured modules to use namespace MySQL::Diff::*
          - restructured modules to use better OO style inferface
          - Utils.pm now only contains debug handling
          - added support for more recent MySQL dumps
          - added more documentation
          - added more tests
        - merged changes by Alexandr Ciornii
          - depend on Perl 5.5.3
          - remove lib/MySQL/.cvsignore
          - fix .gitignore
          - upgrade Makefile.PL and Build.PL

0.33    8th May 2003
        - see ChangeLog.OLD for previous changes.


Makefile.PL
#!/usr/bin/perl -w

use strict;

require 5.006; #use_base_exporter in MySQL::Diff::Utils is usable with 5.6

use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.
WriteMakefile1(
    NAME     => 'MySQL::Diff',
    #KEYWORDS    => [ qw/mysql diff compare schema tables structure database/ ],
    LICENSE  => 'perl',
    META_ADD => {
        resources => {
            license => [ 'http://dev.perl.org/licenses/' ],
            homepage => 'http://adamspiers.org/computing/mysqldiff/',
            bugtracker => {
                web    => 'http://rt.cpan.org/Public/Dist/Display.html?Name=MySQL-Diff',
                mailto => 'mysqldiff@adamspiers.org',
            },
            repository => {
                url  => 'git:github.com/aspiers/mysqldiff.git',
                web  => 'http://github.com/aspiers/mysqldiff',
                type => 'git',
            },
        },
    },
    MIN_PERL_VERSION => '5.006',
    PREREQ_PM => {
        'Carp'        => 0,
        'File::Slurp' => 0,
        'IO::File'    => 0,
    },
    EXE_FILES    => [ 'bin/mysqldiff' ],
    VERSION_FROM => 'lib/MySQL/Diff.pm', # finds $VERSION
    clean => { FILES => "t/logs/ debug.log _build/ Build " },
);

sub WriteMakefile1 {  #Written by Alexandr Ciornii, version 0.21. Added by eumm-upgrade.
    my %params = @_;
    my $eumm_version = $ExtUtils::MakeMaker::VERSION;
    $eumm_version = eval $eumm_version;
    die "EXTRA_META is deprecated" if exists $params{EXTRA_META};
    die "License not specified" if not exists $params{LICENSE};
    if ($params{BUILD_REQUIRES} and $eumm_version < 6.5503) {
        #EUMM 6.5502 has problems with BUILD_REQUIRES
        $params{PREREQ_PM}={ %{$params{PREREQ_PM} || {}} , %{$params{BUILD_REQUIRES}} };
        delete $params{BUILD_REQUIRES};
    }
    delete $params{CONFIGURE_REQUIRES} if $eumm_version < 6.52;
    delete $params{MIN_PERL_VERSION} if $eumm_version < 6.48;
    delete $params{META_MERGE} if $eumm_version < 6.46;
    delete $params{META_ADD} if $eumm_version < 6.46;
    delete $params{LICENSE} if $eumm_version < 6.31;
    delete $params{AUTHOR} if $] < 5.005;
    delete $params{ABSTRACT_FROM} if $] < 5.005;
    delete $params{BINARY_LOCATION} if $] < 5.005;

    WriteMakefile(%params);
}

